include/tvm/task_dynamics/Clamped.h

```diff
@@ -80,66 +54,79 @@ namespace tvm::task_dynamics
+
     /** Constructor with \f$ b_{min} = -b_{max}\f$  (vector version).
       *
-      * \param innerTaskDynamics The task dynamics to clamp.
       * \param max The maximum value that a component of \f$ e_c^{(k)*} \f$ can
       *        have, in absolute value (\f$ b_{max}\f$).
+      * \param args These are forwarded to the TD constructor
       */
-    Clamped(const TD& innerTaskDynamics, const VectorConstRef& max);
-    /** Constructor with \f$ b_{min} = -b_{max}\f$  (vector version).
+    template<typename ... Args>
+    Clamped(const VectorConstRef& max, Args&& ... args);
+
+    /** Constructor with \f$b_{min}\f$ and \f$b_{max}\f$  (vector version).
       *
-      * \param innerTaskDynamics The task dynamics to clamp.
-      * \param max The maximum value that a component of \f$ e_c^{(k)*} \f$ can
-      *        have (\f$ b_{max}\f$).
-      * \param min The minimum value that a component of \f$ e_c^{(k)*} \f$ can
-      *        have (\f$ b_{min}\f$). We need \f$ b_{min} \leq 0 \leq b_{max}\f$.
+      * \param minMax The minimum (\f$b_{min}\f$) and maximum (\f$b_{max}\f$)
+      *               value that a component of \f$ e_c^{(k)*}\f$ can have. We require that
+      *               \f$b_{min} \leq 0 \leq b_{max}\f$.
+      * \param args These are forwarded to the TD constructor
       */
-    Clamped(const TD& innerTaskDynamics, const VectorConstRef& min, const VectorConstRef& max);
+    template<typename ... Args>
+    Clamped(const std::pair<VectorConstRef, VectorConstRef>& minMax, Args&& ... args);
+
+    ~Clamped() override = default;
 
   protected:
-    std::unique_ptr<abstract::TaskDynamicsImpl> impl_(FunctionPtr f, constraint::Type t, const Eigen::VectorXd& rhs) const override;
-    Order order_() const override;
+    std::unique_ptr<abstract::TaskDynamicsImpl> impl_(FunctionPtr f, constraint::Type t, const Eigen::VectorXd& rhs) const override
+    {
+      return TD::template impl_<Impl>(f, t, rhs, min_, max_);
+    }
+
+    COMPOSABLE_TASK_DYNAMICS_DERIVED_FACTORY(TD, min_, max_)
 
   private:
-    TD innerTaskDynamics_;
     Bounds min_;
     Bounds max_;
   };
 
   template<class TD, class TDImpl>
-  inline Clamped<TD, TDImpl>::Clamped(const TD& innerTaskDynamics, double max)
-    : Clamped<TD, TDImpl>(innerTaskDynamics, -max, max)
+  template<typename ... Args>
+  inline Clamped<TD, TDImpl>::Clamped(double max, Args&& ... args)
+    : Clamped<TD, TDImpl>({-max, max}, std::forward<Args>(args)...)
   {
   }
 
   template<class TD, class TDImpl>
-  inline Clamped<TD, TDImpl>::Clamped(const TD& innerTaskDynamics, double min, double max)
-    : innerTaskDynamics_(innerTaskDynamics)
-    , min_(min)
-    , max_(max)
+  template<typename ... Args>
+  inline Clamped<TD, TDImpl>::Clamped(const std::pair<double, double> & minMax, Args&& ... args)
+    : TD(std::forward<Args>(args)...),
+      min_(minMax.first)
+    , max_(minMax.second)
   {
+    const auto & min = minMax.first;
+    const auto & max = minMax.second;
```

Comment:
`const auto& [min, max] = minMax` ?

Reply:
Thanks, I had forgotten this was possible in C++17 :)

Fixed in 4142dd1a (same with the other constructor)